src/progress.h

```diff
@@ -26,10 +26,20 @@ static inline bool HasModalProgress()
 	return _in_modal_progress;
 }
 
+/**
+ * Check if we can use a thread for modal progress.
+ * @return Threading usable?
+ */
+static inline bool UseThreadedModelProgress()
```

Comment:
I wonder how useful this is, given not everywhere else checks for usable threads? (I only see this used for newgrf scan & worldgen) Should probably be all or nothing, and abort if the threads fail

Reply:
Constructing a mutex must succeed, locking a mutex may fail. The mutexes for model progress are locked right in `main`, if that fails it's instant exit. For the real existing C++11 implementations, mutex lock fail will only occur if threading at all is unusable.

The other threading sites try to create their threads first and fall back to a non-threaded implementation. We never got any bug reports that creating a thread succeeded whilst the subsequent mutex lock failed, and I don't see any reason that would change with this PR.